test-lib.sh

```diff
@@ -311,6 +311,49 @@ EOF
   : "  Success!"
 }
 
+# ct_check_exec_env_vars
+# --------------------
+# Checks if all relevant environment variables from `docker run`
+# can be found in `docker exec` as well.
+# Uses: $CID_FILE_DIR - path to directory containing cid_files
+# Uses: $IMAGE_NAME - name of the image being tested
+function ct_check_exec_env_vars() {
+  local tmpdir exec_envs run_envs cid old_IFS
+  local var_name stripped filtered_envs
+  tmpdir=$(mktemp -d)
+  CID_FILE_DIR=${CID_FILE_DIR:-$(mktemp -d)}
+  # Get environment variables from `docker run`
+  docker run --rm "$IMAGE_NAME" /bin/bash -c "env" | sort > "$tmpdir/run_envs"
+  # Get environment variables from `docker exec`
+  ct_create_container "test_exec_envs" bash -c "sleep 1000" >/dev/null
+  cid=$(ct_get_cid "test_exec_envs")
+  docker exec "$cid" env | sort > "$tmpdir/exec_envs"
+  # Filter out variables we are not interested in
+  # Always check X_SCLS, ignore PWD
+  # Check variables from `docker run` that have alternative paths inside (/opt/rh, /opt/app-root)
+  exec_envs=$(cat "$tmpdir/exec_envs")
+  run_envs=$(grep "^X_SCLS=\|/opt/rh\|/opt/app-root" "$tmpdir/run_envs" | grep -v "^PWD=")
+  for variable in $run_envs; do
```

Comment:
While it's very unlikely in the container, where the `env` is under our control, this would not work great if there is a whitespace in the path. I think it might be better without the `$run_envs` variable, but using `while read -r variable ; do ... done <<<$(grep "^X_S...)`.

Reply:
Good catch and I agree, let us go for the safer variant.